Allow window reloading to be turned off

henrahmagix opened the pull request:
Also fixes to current tests as well as new tests.

@incuna/js Please merge, ta!

henrahmagix committed:
Allow reloading to be turned off via config

Also use `windowReload` service everywhere.

henrahmagix committed:
Refresh languageId in template when changing

henrahmagix committed:
Fix controller tests and test changeLanguage()

The tests were using choices that are also the default set in
`languageSelectConfig` so it wasn't apparent if the mocks were actually
being used.

Also all spies must be set _before_ creating the controller, else they
won't be in use.

henrahmagix committed:
Test reloadOnChange config and err for some values

henrahmagix committed:
Test windowReload

henrahmagix committed:
Update changelog

henrahmagix committed:
Fix setReloadOnChange not.toThrowError on undefined

wytrych commented on app/scripts/src/config.js:
```diff
@@ -28,5 +32,12 @@ module.provider('languageSelectConfig', function () {
+                reloadOnChange = false;
+            } else if (typeof value !== 'undefined') {
+                throw new Error('Cannot set reloadOnChange to ' + JSON.stringify(value) + ' [' + typeof value + ']');
+            }
```
I don't think it's useful to allow empty calls on this function, it would be easier to do a type check:

```js
if (typeof value !== 'boolean') {
    throw new Error('Value should be a boolean')
}
reloadOnChange  = value;
```

henrahmagix replied:
Done in f25f71c

wytrych commented on tests/storage-spec.js:
```diff
@@ -1,5 +1,41 @@
 import '../app/scripts/src/main';
 
+describe('windowReload', function () {
```
Should this live in a separate file?

henrahmagix replied:
Perhaps, but it's not in a separate file in the src. Don't think it matters much since it's quite small

wytrych commented:
👍

henrahmagix committed:
Fully mock cookieHandler in tests to avoid cookies

Otherwise tests are affected by lingering cookies. Only use the browser
cookies for the tests on `document.cookie`, otherwise use the mocks.

henrahmagix committed:
Fix tests on windowReload call counts

We can't expect it to not be called upon app bootstrap, so instead test
the number of times it's been called has only increased by 1.

henrahmagix committed:
Fix grunt failing if swig templates are cleaned

And warn the dev

henrahmagix committed:
Auto-fix missing files for ngtemplates config.

Run a child process to compile them, then carry on with grunt.

henrahmagix committed:
Simplify config setReloadOnChange arg checking.

henrahmagix commented:
@wytrych Updated

wytrych commented:
👍

henrahmagix committed:
Test fix for missing files for ngtemplates config

See 889a2ee070532735734cef23fa8145f4d7fa5857

henrahmagix commented:
Thanks @wytrych!

henrahmagix commented:
Released as [`2.2.0`](https://github.com/incuna/angular-language-select/releases/tag/2.2.0)
